impl/src/main/java/com/okta/sdk/impl/client/DefaultClientBuilder.java

```diff
@@ -329,8 +392,55 @@ public ClientBuilder setOrgUrl(String baseUrl) {
         return this;
     }
 
+    @Override
+    public ClientBuilder setAuthorizationMode(String authorizationMode) {
```

Comment:
If it's only ever "PrivateKey" do we need this property at all? Just use the existing AuthorizationScheme attribute?
Or nothing at all, just look for the token or privateKey and make the determination.

Reply:
I thought of the same for the sake of simplicity. But the design document mentions the use of authorizationMode attribute to distinguish between OAUTH2 and SSWS (default) flow. To stick with the design document, I had to make use of this parameter.

Reply:
@arvindkrishnakumar-okta 
I would suggest the following, deprecate the current `setAuthenticationScheme` and replace it with this one and make the argument the `AuthenticationScheme` enum, (you would need to change the `OAUTH2` value to PrivateKey` to match the current spec.

(that would probably remove some of the confusion I'm seeing with multiple properties that do the same thing)

NOTE: the `setAuthenticationScheme` is a hold over from the original Stormpath SDK
https://github.com/stormpath/stormpath-sdk-java/blob/1.1.x/api/src/main/java/com/stormpath/sdk/client/ClientBuilder.java#L328

Now that this functionality is in the SDK spec the old method can be deprecated

Reply:
@bdemers noted. Please see my other comment on this subject.